src/core/action/channels/ObtainRGBChannelAction.java

```diff
@@ -0,0 +1,57 @@
+package core.action.channels;
+
+import core.repository.ImageRepository;
+import domain.CustomImage;
+import domain.RGBChannel;
+import javafx.scene.image.Image;
+import javafx.scene.image.PixelWriter;
+import javafx.scene.image.WritableImage;
+import javafx.scene.paint.Color;
+
+public class ObtainRGBChannelAction {
+
+    private final ImageRepository repository;
+
+    public ObtainRGBChannelAction(ImageRepository repository) {
+        this.repository = repository;
+    }
+
+    public Image execute (RGBChannel channel) {
+
+        //Aca habria que usar el ifPresent, pero yo no se bien como se debe hacer, a corregir.
+        CustomImage image = this.repository.get(this.repository.getCurrentImagePath().get());
+
+        int width = image.getWidth();
+        int height = image.getHeight();
+        WritableImage writableImage = new WritableImage(width, height);
+        PixelWriter pixelWriter = writableImage.getPixelWriter();
+
+        for (int x = 0; x < width; x++) {
+            for (int y = 0; y < height; y++) {
+
+                switch (channel) {
+
```

Comment:
Me parece que este switch case tendría que estás por fuera del for para no preguntar constantemente por el valor de channel. Y los for podrían ir en un método para reutilizarlos

Reply:
Entendi. Igual preocuparte por la eficiencia es al pedo, salvo que recorras imagenes de veinte millones de pixeles (cosa que no haremos nunca) no va a hacer diferencia.

Explicame como reutilizarias los for, eso si no lo entendi